### PR TITLE
Add `margin_sd` to gaussian filtering

### PR DESCRIPTION
**Pull request by DradeAW**
Add the option to change the margins for the gaussian filter.

### CODE REVIEW

File: src/spikeinterface/preprocessing/filter_gaussian.py [hunk 1]
```diff
@@ -32,19 +34,23 @@ class GaussianBandpassFilterRecording(BasePreprocessor):
 
     name = "gaussian_bandpass_filter"
 
-    def __init__(self, recording: BaseRecording, freq_min: float = 300.0, freq_max: float = 5000.0):
+    def __init__(
+        self, recording: BaseRecording, freq_min: float = 300.0, freq_max: float = 5000.0, margin_sd: float = 5.0
```
Comment (alejoe91) [inline src/spikeinterface/preprocessing/filter_gaussian.py:38]:
```suggestion
        self, recording: BaseRecording, freq_min: float = 300.0, freq_max: float = 5000.0, margin_sd: float = 6.0
```
?

Comment (DradeAW) [reply src/spikeinterface/preprocessing/filter_gaussian.py:38]: I think 6 by default is too high. Even 5 might be too high and 4 should be completely fine (there's nothing after 4 sd in a Gaussian distribution).

This should help speed  things up a bit.

Comment (DradeAW) [reply src/spikeinterface/preprocessing/filter_gaussian.py:38]: For reference, at 30kHz, a gaussian filter at 300 Hz would take `5*15.92 = 80` ms margins (compared to the 5ms of the "regular filter")

Comment (alejoe91) [reply src/spikeinterface/preprocessing/filter_gaussian.py:38]: Just wanted to point out that there was a mismatch between the default in the init and in the doc! ;)

File: src/spikeinterface/preprocessing/filter_gaussian.py [hunk 2]
```diff
@@ -23,6 +23,8 @@ class GaussianBandpassFilterRecording(BasePreprocessor):
         The lower frequency cutoff for the bandpass filter.
     freq_max: float
         The higher frequency cutoff for the bandpass filter.
+    margin_sd: float, default: 6.0
```
Comment (alejoe91) [inline src/spikeinterface/preprocessing/filter_gaussian.py:26]:
```suggestion
    margin_sd: float, default: 5.0
```

Comment (DradeAW) [reply src/spikeinterface/preprocessing/filter_gaussian.py:26]: Ah right, good catch!

**Review by alejoe91 — APPROVED** [review]

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**